core/new-gui/src/app/dashboard/user/component/user-workflow/ngbd-modal-workflow-executions/ngbd-modal-workflow-executions.component.ts

```diff
@@ -723,4 +710,19 @@ export class NgbdModalWorkflowExecutionsComponent implements OnInit, AfterViewIn
     });
     return processTimeData;
   }
+
+  private updatePaginatedExecutions(): void {
+    this.paginatedExecutionEntries = this.changePaginatedExecutions();
+    this.workflowExecutionsDisplayedList = this.paginatedExecutionEntries;
+    this.fuse.setCollection(this.paginatedExecutionEntries);
+  }
+
+  private handlePaginationAfterDeletingExecutions(): void {
+    this.updatePaginatedExecutions();
+    /* If a current page index has 0 number of execution entries after deletion (e.g., deleting all the executions in the last page),
+     * the following code will decrement the current page index by 1. */
+    if (this.currentPageIndex > 1 && this.paginatedExecutionEntries.length === 0) {
+      this.onPageIndexChange(this.currentPageIndex - 1);
```

Comment:
one minor question: Let's suppose the user deleted all executions on page 1, and it automatically changed the page to page 0. Then will the user be able to select page 1 again? Also, what's the behavior of this if the user deleted all entries on page 0?

Reply:
It is impossible for the user to navigate to page 0, as the page index change feature only works for page index two or higher. This is due to the condition `this.currentPageIndex > 1` in the if statement.

Reply:
got it, thanks! You can merge it now.